aptos-move/framework/aptos-framework/sources/aptos_governance.spec.move

```diff
@@ -175,61 +216,202 @@ spec aptos_framework::aptos_governance {
+        let is_voting_closed_3 = is_voting_period_over || can_be_resolved_early_3;
+        let proposal_state_successed_3 = is_voting_closed_3 && proposal.yes_votes > new_proposal_no_votes_1 &&
+                                         proposal.yes_votes + new_proposal_no_votes_1 >= proposal.min_vote_threshold;
+        // post state
+        let post can_be_resolved_early = option::spec_is_some(proposal.early_resolution_vote_threshold) &&
+                                    (post_proposal.yes_votes >= early_resolution_threshold ||
+                                     post_proposal.no_votes >= early_resolution_threshold);
+        let post is_voting_closed = is_voting_period_over || can_be_resolved_early;
+        let post proposal_state_successed = is_voting_closed && post_proposal.yes_votes > post_proposal.no_votes &&
+                                         post_proposal.yes_votes + post_proposal.no_votes >= proposal.min_vote_threshold;
+        // verify add_approved_script_hash(proposal_id)
+        let execution_hash = proposal.execution_hash;
+        let post post_approved_hashes = global<ApprovedExecutionHashes>(@aptos_framework);
+
+        // Due to the complexity of the success state, the validation of 'borrow_global_mut<ApprovedExecutionHashes>(@aptos_framework);' is discussed in four cases.
+        aborts_if allow_validator_set_change &&
+            if (should_pass) {
+                proposal_state_successed_0 && !exists<ApprovedExecutionHashes>(@aptos_framework)
+            } else {
+                proposal_state_successed_1 && !exists<ApprovedExecutionHashes>(@aptos_framework)
+            };
+        aborts_if !allow_validator_set_change &&
+            if (should_pass) {
+                proposal_state_successed_2 && !exists<ApprovedExecutionHashes>(@aptos_framework)
+            } else {
+                proposal_state_successed_3 && !exists<ApprovedExecutionHashes>(@aptos_framework)
+            };
+        ensures proposal_state_successed ==> simple_map::spec_contains_key(post_approved_hashes.hashes, proposal_id) &&
+                                             simple_map::spec_get(post_approved_hashes.hashes, proposal_id) == execution_hash;
     }
 
     spec add_approved_script_hash(proposal_id: u64) {
         use aptos_framework::chain_status;
-        // TODO: The variable `proposal_state` is the return value of the function `voting::get_proposal_state`.
-        // The calling level of `voting::get_proposal_state` is very deep,
-        // so the value of `proposal_state` cannot be obtained in the spec,
-        // and the `aborts_if` of `proposal_state` cannot be written.
-        // Can't cover all aborts_if conditions
-        pragma aborts_if_is_partial;
 
         requires chain_status::is_operating();
-        aborts_if !exists<ApprovedExecutionHashes>(@aptos_framework);
+        include Schema_add_approved_script_hash;
```

Comment:
Could you change the schema name to something like `SchemaAddApprovedScriptHash`?